### ISSUE TITLE
compilation error with mysql

### ISSUE THREAD

**Issue by vincefn** [vincefn/objcryst#48]
There is a compilation error with recent mysql (e.g. ubuntu 20.04, debian bullseye):
```
g++ -O3 -w -ffast-math -fstrict-aliasing -pipe -fomit-frame-pointer -funroll-loops -ftree-vectorize -DHAVE_SSE_MATHFUN -DUSE_SSE2 -march=native -D__FOX_COD__ -I/usr/include/mysql -I/include -I/home/esrf/favre/dev/Fox-2017-06-28/Fox/src/../../ObjCryst/.. -I/home/esrf/favre/dev/Fox-2017-06-28/Fox/src/../../static-libs/include -DOBJCRYST_GL -I/usr/X11R6/include -IGL -DHAVE_GLUT -D__WX__CRYST__ `/home/esrf/favre/dev/Fox-2017-06-28/Fox/src/../../ObjCryst/../static-libs/bin/wx-config --unicode=yes --cxxflags` -DHAVE_FFTW -DREAL=float  -c Fox.cpp -o Fox.o
In file included from Fox.cpp:70:
/usr/include/mysql/mysql.h:470:6: error: use of enum ‘net_async_status’ without previous declaration
  470 | enum net_async_status STDCALL mysql_real_connect_nonblocking(
      |      ^~~~~~~~~~~~~~~~
/usr/include/mysql/mysql.h:474:6: error: use of enum ‘net_async_status’ without previous declaration
  474 | enum net_async_status STDCALL mysql_send_query_nonblocking(
      |      ^~~~~~~~~~~~~~~~
/usr/include/mysql/mysql.h:476:6: error: use of enum ‘net_async_status’ without previous declaration
  476 | enum net_async_status STDCALL mysql_real_query_nonblocking(
      |      ^~~~~~~~~~~~~~~~

```

**Comment by vincefn** [comment]
This error occurs e.g. on ubuntu 20.04, likely a change in mysql, C++ compliance, or a conflict of variables ?

One workaround should be to disable the COD access by compiling with the 'cod=0' option, e.g.:

make -j4 Fox cod=0

Ultimately the solution should be to remove the dependency to mysql by using only web queries.

**Comment by vincefn** [comment]
Fixed by 973ba384d8f024e82effe442a26a5711eb83b1c7
